Change back link arrow to chevron

RobBond-GDS opened the pull request:
Update styling of back link component

https://trello.com/c/ZYJ7nnYt/21-make-back-icon-more-consistent-on-mobiles

## What
Update styling on back link component and increase touch target on mobile devices

## Why
Back links use an inconsistent icon from the breadcrumbs, The link also has a small touch target on mobiles. We should have consistent design patterns across GOV.UK. 

## Visual Changes
![Screenshot 2020-02-17 at 13 43 32](https://user-images.githubusercontent.com/54625020/74659295-f6bb5e80-518b-11ea-926d-9ee80ae44c7a.png)
Before change - mobile

![Screenshot 2020-02-17 at 13 44 11](https://user-images.githubusercontent.com/54625020/74659311-fde26c80-518b-11ea-8013-f655945b0347.png)
Before Change - desktop


![Screenshot 2020-02-17 at 13 42 50](https://user-images.githubusercontent.com/54625020/74659322-033fb700-518c-11ea-956b-9818de91e0be.png)
After change - mobile

![Screenshot 2020-02-17 at 13 44 54](https://user-images.githubusercontent.com/54625020/74659330-06d33e00-518c-11ea-9fd9-f73b280939ee.png)
After Change - desktop
<!-- If the change results in visual changes, show a before and after -->

maxgds commented:
I'm not convinced we should be diverging from the design system at this level. We're in a world where many things are inconsistent and we're trying to adopt the design system as fully as possible to overcome that so this feels like a backwards step. If the touch area isn't good enough I think it should be fixed at source rather than over-ridden. If you haven't already done so I'd suggest talking to the team about it: https://design-system.service.gov.uk/components/back-link/

injms commented:
From the looks of the [talk around the back link component itself](https://github.com/alphagov/govuk-design-system-backlog/issues/32), it would seem that the Home Office have done some research around improvements to the back button. There's also this [discussion about harmonising the design of the back link and breadcrumbs](https://github.com/alphagov/govuk-design-system-backlog/issues/33#issuecomment-465957906). I agree with @maxgds - it's better to try and fix this at source.

RobBond-GDS commented:
Thanks for the feedback, as it stands the mobile team has already had
conversations with the design team and the tasks that have been proposed
have been given the go-ahead by them as far as implementing on GOV.UK is
concerned. This is with a view to feeding back into the design system at a
later date.

On Mon, 17 Feb 2020 at 15:16, Ian James <notifications@github.com> wrote:

> From the looks of the talk around the back link component itself
> <https://github.com/alphagov/govuk-design-system-backlog/issues/32>, it
> would seem that the Home Office have done some research around improvements
> to the back button. There's also this discussion about harmonising the
> design of the back link and breadcrumbs
> <https://github.com/alphagov/govuk-design-system-backlog/issues/33#issuecomment-465957906>.
> I agree with @maxgds <https://github.com/maxgds> - it's better to try and
> fix this at source.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/alphagov/govuk_publishing_components/pull/1299?email_source=notifications&email_token=ANAYF7CTWWOTK3V6QNXSVDDRDKS6LA5CNFSM4KWSCPBKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEL6YQYA#issuecomment-587040864>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANAYF7F3EDIPWDU2CMQ5U2DRDKS6LANCNFSM4KWSCPBA>
> .
>

maxgds commented:
If this is to be merged in, I think you need to give the chevron some further attention. On your screenshot the < seems to be misaligned vertically (it looks too low) on the mobile view and when I view it in chrome locally with dev tools responsive it's also misaligned but in the opposite direction by a couple of pixels.

RobBond-GDS commented:
> If this is to be merged in, I think you need to give the chevron some further attention. On your screenshot the < seems to be misaligned vertically (it looks too low) on the mobile view and when I view it in chrome locally with dev tools responsive it's also misaligned but in the opposite direction by a couple of pixels.

andysellick commented on app/assets/stylesheets/govuk_publishing_components/components/_back-link.scss:
```diff
@@ -1 +1,45 @@
+  }
+
+  &:before {
+    content: "";
```
Is it not possible to use the govuk-shape mixin here? e.g.

`@include govuk-shape-arrow($direction: left, $base: 10px, $height: 6px);`

RobBond-GDS replied:
This would give a solid triangle, design calls for a left-facing chevron

RobBond-GDS commented:
I shall check its alignment with the breadcrumb component as that would be the pattern it is to compliment

andysellick commented on app/assets/stylesheets/govuk_publishing_components/components/_back-link.scss:
```diff
@@ -1 +1,45 @@
+}
+
+@include govuk-media-query($until: tablet) {
+
```
Nitpick: empty line.

RobBond-GDS replied:
have removed the empty line

andysellick commented on app/assets/stylesheets/govuk_publishing_components/components/_back-link.scss:
```diff
@@ -1 +1,45 @@
+@include govuk-media-query($until: tablet) {
+
+  .govuk-back-link {
+    position: relative;
```
What's this positioning for? Looks like it's already defined in the inspector.

RobBond-GDS replied:
This applies the same pseudo-element to increase the touch target after the link as used in the breadcrumb PR

andysellick replied:
What I mean is that position: relative is already set on this element - it doesn't need setting again.

RobBond-GDS replied:
See what you mean, have removed duplicate

RobBond-GDS commented:
![Slice](https://user-images.githubusercontent.com/54625020/74673251-c7feb180-51a6-11ea-8f39-07b9890bc267.png)

Took screenshot of both breadcrumb and back link, overlaid on on the other. As you can see they chevron is aligned the same on both and same size

vanitabarrett commented on app/assets/stylesheets/govuk_publishing_components/components/_back-link.scss:
```diff
@@ -1 +1,40 @@
+  }
+}
+
+@include govuk-media-query($until: tablet) {
```
This may be a question for the team, but I wonder if we want the larger touch target to apply all the time including on desktop? I'm not sure what the recommendations are, but it seems feasible that someone could be viewing the site in desktop view but on a touchscreen device.

RobBond-GDS replied:
Asked PM and response was "That makes sense to me - would that have any adverse effect on the look of anything around the Back link on desktop? If not, happy to go with that change for desktop too. " 

As it wouldn't affect desktop have removed media query so it was just on mobile

RobBond-GDS commented on app/assets/stylesheets/govuk_publishing_components/components/_back-link.scss:
```diff
@@ -1 +1,40 @@
+    position: absolute;
+    top: 0;
+    bottom: 1px;
+    left: 3.31px;
```
so the chevron remains in the bounds of the focused state

vanitabarrett replied:
`3px` seems to work ok?

RobBond-GDS commented on app/assets/stylesheets/govuk_publishing_components/components/_back-link.scss:
```diff
@@ -1 +1,40 @@
+    width: 7px;
+    height: 7px;
+    -webkit-transform: rotate(225deg);
+    -ms-transform: rotate(2255deg);
```
amended typo

vanitabarrett commented on CHANGELOG.md:
```diff
@@ -7,8 +7,10 @@
   useful summary for people upgrading their application, not a replication
   of the commit log.
 
+
```
Can you remove this additional space and squash please? I think this PR needs another rebase too as someone has just versioned the gem.

RobBond-GDS commented:
Squashed and rebased

On Wed, 19 Feb 2020 at 09:41, Vanita Barrett <notifications@github.com>
wrote:

> *@vanitabarrett* commented on this pull request.
> ------------------------------
>
> In CHANGELOG.md
> <https://github.com/alphagov/govuk_publishing_components/pull/1299#discussion_r381176879>
> :
>
> > @@ -7,8 +7,10 @@
>    useful summary for people upgrading their application, not a replication
>    of the commit log.
>
> +
>
> Can you remove this additional space and squash please? I think this PR
> needs another rebase too as someone has just versioned the gem.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/alphagov/govuk_publishing_components/pull/1299?email_source=notifications&email_token=ANAYF7DIVRAUM37GFMEQNFDRDT5C3A5CNFSM4KWSCPBKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCWB6CCY#pullrequestreview-360964363>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANAYF7EHOAOFVK7GTZW54GLRDT5C3ANCNFSM4KWSCPBA>
> .
>
